File: src/sdk-reference/cpp/1/realtime/subscribe/snippets/user-notifications.cpp
kuzzleio::NotificationListener listener =
  [](const std::shared_ptr<kuzzleio::notification_result> &notification) {
    std::cout << "Currently " << notification->result->count << " users in the room" << std::endl;

    std::cout << notification->volatiles << std::endl;
    // "{ "username": "nina vkote" }"
  };

try {
  // Subscription to notifications and notifications when user join or leave
  const char *filters = R"({ "exists": "name" })";
  kuzzleio::room_options options;
  options.users = "all";

  kuzzle->realtime->subscribe(
    "nyc-open-data",
    "yellow-taxi",
    filters,
    &listener,
    options);

  // Instantiates a second kuzzle client:
  //  multiple subscriptions made by the same user
  //  will not trigger "new user" notifications
  kuzzleio::WebSocket* ws2 = new kuzzleio::WebSocket("kuzzle");
  kuzzleio::Kuzzle* kuzzle2 = new kuzzleio::Kuzzle(ws2);
  kuzzle2->connect();

  // Set some volatile data
  kuzzleio::room_options options2;
  options2.volatiles = R"({ "username": "nina vkote" })";

  // Subscribe to the same room with the second client
  kuzzle2->realtime->subscribe(
    "nyc-open-data",
    "yellow-taxi",
    filters,
    &listener,
    options2);
} catch (kuzzleio::KuzzleException &e) {
  std::cerr << e.what() << std::endl;
}

File: src/sdk-reference/csharp/1/realtime/subscribe/snippets/user-notifications.cs
NotificationListener listener =
  [](const notification_result *notification) {
    Console.WriteLine("Currently " + notification.result.count + " users in the room");

    Console.WriteLine(notification.volatiles);
    // "{ "username": "nina vkote" }"
  };

try {
  // Subscription to notifications and notifications when user join or leave
  const char *filters = @"{ ""exists"": ""name"" }";
  room_options options;
  options.users = "all";

  kuzzle.realtime.subscribe(
    "nyc-open-data",
    "yellow-taxi",
    filters,
    listener,
    options);

  // Instantiates a second kuzzle client:
  //  multiple subscriptions made by the same user
  //  will not trigger "new user" notifications
  WebSocket* ws2 = new WebSocket("kuzzle");
  Kuzzle* kuzzle2 = new Kuzzle(ws2);
  kuzzle2.connect();

  // Set some volatile data
  room_options options2;
  options2.volatiles = @"{ ""username"": ""nina vkote"" }";

  // Subscribe to the same room with the second client
  kuzzle2.realtime.subscribe(
    "nyc-open-data",
    "yellow-taxi",
    filters,
    listener,
    options2);
} catch (KuzzleException e) {
  Console.Error.WriteLine(e.Message());
}
